The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8113194888/job/22176028738) [(plain)](https://github.com/rust-lang/rust/commit/c5dda95af7af1cffb96ae657df2e8daa4116ccad/checks/22176028738/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=ChrisDenton
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_fda639d7-aaee-4fff-8a3a-7057f7e5f086
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=f796019acb415d7bbeca359abb894fcb60dda0e8
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_fda639d7-aaee-4fff-8a3a-7057f7e5f086
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_fda639d7-aaee-4fff-8a3a-7057f7e5f086
GITHUB_TRIGGERING_ACTOR=ChrisDenton
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/121856/merge
GITHUB_WORKFLOW_SHA=f796019acb415d7bbeca359abb894fcb60dda0e8
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_20_X64=/opt/hostedtoolcache/go/1.20.14/x64
---
   Compiler: "MIRI_ENV_VAR_TEST"="0" "MIRI_TEMP"="/tmp" /checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-msvc" "--out-dir" OUT_DIR


FAILED TEST: tests/fail/intrinsics/uninit_uninhabited_type.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-msvc" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui/tests/fail/intrinsics" "tests/fail/intrinsics/uninit_uninhabited_type.rs" "--edition" "2021"
error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/intrinsics/uninit_uninhabited_type.stderr` to the actual output
Error: 
   0: tests failed
   0: tests failed

Backtrace omitted. Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
+++ <stderr output>
 thread 'main' panicked at RUSTLIB/core/src/panicking.rs:LL:CC:
 aborted execution: attempted to instantiate uninhabited type `!`
... 3 lines skipped ...
---
   |
   = note: inside `std::sys::pal::windows::abort_internal` at /checkout/library/std/src/sys/pal/windows/mod.rs:340:5: 340:30
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:798:9: 798:37
   = note: inside closure at /checkout/library/std/src/panicking.rs:649:13: 655:14
   = note: inside `std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys_common/backtrace.rs:171:18: 171:21
   = note: inside `core::panicking::panic_nounwind` at /checkout/library/core/src/panicking.rs:155:5: 155:91
note: inside `main`
  --> tests/fail/intrinsics/uninit_uninhabited_type.rs:10:22
   |
   |
LL |     let _ = unsafe { std::mem::uninitialized::<!>() };

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to 1 previous error
error: aborting due to 1 previous error


full stdout:



FAILED TEST: tests/fail/intrinsics/zero_fn_ptr.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-msvc" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui/tests/fail/intrinsics" "tests/fail/intrinsics/zero_fn_ptr.rs" "--edition" "2021"
error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/intrinsics/zero_fn_ptr.stderr` to the actual output
--- tests/fail/intrinsics/zero_fn_ptr.stderr
+++ <stderr output>
+++ <stderr output>
 thread 'main' panicked at RUSTLIB/core/src/panicking.rs:LL:CC:
 aborted execution: attempted to zero-initialize type `fn()`, which is invalid
   --> RUSTLIB/std/src/sys/pal/PLATFORM/mod.rs:LL:CC
    |
-LL |     ABORT();
+LL |     core::intrinsics::abort()
---


full stderr:
thread 'main' panicked at /checkout/library/core/src/panicking.rs:155:5:
aborted execution: attempted to zero-initialize type `fn()`, which is invalid
thread caused non-unwinding panic. aborting.
error: abnormal termination: the program aborted execution
##[error]  --> /checkout/library/std/src/sys/pal/windows/mod.rs:340:5
   |
   |
LL |     core::intrinsics::abort()
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^ the program aborted execution
   |
   = note: inside `std::sys::pal::windows::abort_internal` at /checkout/library/std/src/sys/pal/windows/mod.rs:340:5: 340:30
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:798:9: 798:37
   = note: inside closure at /checkout/library/std/src/panicking.rs:649:13: 655:14
   = note: inside `std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys_common/backtrace.rs:171:18: 171:21
   = note: inside `core::panicking::panic_nounwind` at /checkout/library/core/src/panicking.rs:155:5: 155:91
note: inside `main`
  --> tests/fail/intrinsics/zero_fn_ptr.rs:9:22
   |
   |
LL |     let _ = unsafe { std::mem::zeroed::<fn()>() };

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to 1 previous error
error: aborting due to 1 previous error


full stdout:



FAILED TEST: tests/fail/terminate-terminator.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-msvc" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui/tests/fail" "tests/fail/terminate-terminator.rs" "-Zmir-opt-level=3" "-Zinline-mir-hint-threshold=1000" "--edition" "2021"
error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/terminate-terminator.stderr` to the actual output
--- tests/fail/terminate-terminator.stderr
+++ <stderr output>
+++ <stderr output>
 warning: You have explicitly enabled MIR optimizations, overriding Miri's default which is to completely disable them. Any optimizations may hide UB that Miri would otherwise detect, and it is not necessarily possible to predict what kind of UB will be missed. If you are enabling optimizations to make Miri run faster, we advise using cfg(miri) to shrink your workload instead. The performance benefit of enabling MIR optimizations is usually marginal at best.
... 8 lines skipped ...
   --> RUSTLIB/std/src/sys/pal/PLATFORM/mod.rs:LL:CC
    |
-LL |     ABORT();
---
 


full stderr:
warning: You have explicitly enabled MIR optimizations, overriding Miri's default which is to completely disable them. Any optimizations may hide UB that Miri would otherwise detect, and it is not necessarily possible to predict what kind of UB will be missed. If you are enabling optimizations to make Miri run faster, we advise using cfg(miri) to shrink your workload instead. The performance benefit of enabling MIR optimizations is usually marginal at best.
thread 'main' panicked at tests/fail/terminate-terminator.rs:21:5:
explicit panic
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /checkout/library/core/src/panicking.rs:155:5:
thread 'main' panicked at /checkout/library/core/src/panicking.rs:155:5:
panic in a function that cannot unwind
stack backtrace:
   0:   0xbee478 - std::backtrace_rs::backtrace::miri::trace_unsynchronized::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:99:5
   1:   0xbee01a - std::backtrace_rs::backtrace::miri::trace::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:62:14
   2:   0xbec997 - std::backtrace_rs::backtrace::trace_unsynchronized::<{closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
   3:   0xbc35bb - std::sys_common::backtrace::_print_fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:68:5
   4:   0xbc22f0 - <std::sys_common::backtrace::_print::DisplayBacktrace as std::fmt::Display>::fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:44:22
---
  12:   0x19a092 - std::panicking::rust_panic_with_hook
                       at /checkout/library/std/src/panicking.rs:779:13
  13:   0x198d05 - std::panicking::begin_panic_handler::{closure#0}
                       at /checkout/library/std/src/panicking.rs:649:13
  14:   0xbc4425 - std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>
  15:   0x198c02 - std::panicking::begin_panic_handler
                       at /checkout/library/std/src/panicking.rs:645:5
  16:   0x15db00 - core::panicking::panic_nounwind_fmt::runtime
                       at /checkout/library/core/src/panicking.rs:110:18
---
  28:    0x806a3 - std::panic::catch_unwind::<&dyn std::ops::Fn() -> i32 + std::marker::Sync + std::panic::RefUnwindSafe, i32>
                       at /checkout/library/std/src/panic.rs:146:14
  29:   0x176184 - std::rt::lang_start_internal::{closure#2}
                       at /checkout/library/std/src/rt.rs:148:48
  30:   0x197ffa - std::panicking::r#try::do_call::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
                       at /checkout/library/std/src/panicking.rs:552:40
  31:   0x19798b - std::panicking::r#try::<isize, {closure@std::rt::lang_start_internal::{closure#2}}>
                       at /checkout/library/std/src/panicking.rs:516:19
  32:    0x806c1 - std::panic::catch_unwind::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
  33:   0x176189 - std::rt::lang_start_internal
                       at /checkout/library/std/src/rt.rs:148:20
  34:   0x17637c - std::rt::lang_start::<()>
                       at /checkout/library/std/src/rt.rs:165:17
---
   |
   = note: inside `std::sys::pal::windows::abort_internal` at /checkout/library/std/src/sys/pal/windows/mod.rs:340:5: 340:30
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:798:9: 798:37
   = note: inside closure at /checkout/library/std/src/panicking.rs:649:13: 655:14
   = note: inside `std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys_common/backtrace.rs:171:18: 171:21
   = note: inside `core::panicking::panic_nounwind` at /checkout/library/core/src/panicking.rs:155:5: 155:91
   = note: inside `core::panicking::panic_cannot_unwind` at /checkout/library/core/src/panicking.rs:241:5: 241:61
note: inside `has_cleanup`
  --> tests/fail/terminate-terminator.rs:19:1
---



FAILED TEST: tests/fail/unwind-action-terminate.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-msvc" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui/tests/fail" "tests/fail/unwind-action-terminate.rs" "--edition" "2021"
error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/unwind-action-terminate.stderr` to the actual output
--- tests/fail/unwind-action-terminate.stderr
+++ <stderr output>
---
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /checkout/library/core/src/panicking.rs:155:5:
panic in a function that cannot unwind
stack backtrace:
   0:   0xbee333 - std::backtrace_rs::backtrace::miri::trace_unsynchronized::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:99:5
   1:   0xbeded5 - std::backtrace_rs::backtrace::miri::trace::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:62:14
   2:   0xbec852 - std::backtrace_rs::backtrace::trace_unsynchronized::<{closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
   3:   0xbc3476 - std::sys_common::backtrace::_print_fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:68:5
   4:   0xbc21ab - <std::sys_common::backtrace::_print::DisplayBacktrace as std::fmt::Display>::fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:44:22
---
  12:   0x199f4d - std::panicking::rust_panic_with_hook
                       at /checkout/library/std/src/panicking.rs:779:13
  13:   0x198bc0 - std::panicking::begin_panic_handler::{closure#0}
                       at /checkout/library/std/src/panicking.rs:649:13
  14:   0xbc42e0 - std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>
  15:   0x198abd - std::panicking::begin_panic_handler
                       at /checkout/library/std/src/panicking.rs:645:5
  16:   0x15d9bb - core::panicking::panic_nounwind_fmt::runtime
                       at /checkout/library/core/src/panicking.rs:110:18
---
  28:    0x8055e - std::panic::catch_unwind::<&dyn std::ops::Fn() -> i32 + std::marker::Sync + std::panic::RefUnwindSafe, i32>
                       at /checkout/library/std/src/panic.rs:146:14
  29:   0x17603f - std::rt::lang_start_internal::{closure#2}
                       at /checkout/library/std/src/rt.rs:148:48
  30:   0x197eb5 - std::panicking::r#try::do_call::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
                       at /checkout/library/std/src/panicking.rs:552:40
  31:   0x197846 - std::panicking::r#try::<isize, {closure@std::rt::lang_start_internal::{closure#2}}>
                       at /checkout/library/std/src/panicking.rs:516:19
  32:    0x8057c - std::panic::catch_unwind::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
  33:   0x176044 - std::rt::lang_start_internal
                       at /checkout/library/std/src/rt.rs:148:20
  34:   0x176237 - std::rt::lang_start::<()>
                       at /checkout/library/std/src/rt.rs:165:17
---
   |
   = note: inside `std::sys::pal::windows::abort_internal` at /checkout/library/std/src/sys/pal/windows/mod.rs:340:5: 340:30
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:798:9: 798:37
   = note: inside closure at /checkout/library/std/src/panicking.rs:649:13: 655:14
   = note: inside `std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys_common/backtrace.rs:171:18: 171:21
   = note: inside `core::panicking::panic_nounwind` at /checkout/library/core/src/panicking.rs:155:5: 155:91
   = note: inside `core::panicking::panic_cannot_unwind` at /checkout/library/core/src/panicking.rs:241:5: 241:61
note: inside `panic_abort`
  --> tests/fail/unwind-action-terminate.rs:8:1
---



FAILED TEST: tests/fail/panic/double_panic.rs
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-msvc" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui/tests/fail/panic" "tests/fail/panic/double_panic.rs" "--edition" "2021"
error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/panic/double_panic.stderr` to the actual output
--- tests/fail/panic/double_panic.stderr
+++ <stderr output>
---
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at tests/fail/panic/double_panic.rs:10:9:
second
stack backtrace:
   0:   0xbed464 - std::backtrace_rs::backtrace::miri::trace_unsynchronized::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:99:5
   1:   0xbed010 - std::backtrace_rs::backtrace::miri::trace::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:62:14
   2:   0xbeb980 - std::backtrace_rs::backtrace::trace_unsynchronized::<{closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
   3:   0xbc2598 - std::sys_common::backtrace::_print_fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:68:5
   4:   0xbc1356 - <std::sys_common::backtrace::_print::DisplayBacktrace as std::fmt::Display>::fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:44:22
---
  12:   0xa3173b - std::panicking::rust_panic_with_hook
                       at /checkout/library/std/src/panicking.rs:779:13
  13:   0xa303af - std::panicking::begin_panic_handler::{closure#0}
                       at /checkout/library/std/src/panicking.rs:649:13
  14:   0xbc33eb - std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>
  15:   0xa3029f - std::panicking::begin_panic_handler
                       at /checkout/library/std/src/panicking.rs:645:5
  16:   0x15c462 - std::rt::panic_fmt
                       at /checkout/library/core/src/panicking.rs:72:14
---
  26:    0x7f661 - std::panic::catch_unwind::<&dyn std::ops::Fn() -> i32 + std::marker::Sync + std::panic::RefUnwindSafe, i32>
                       at /checkout/library/std/src/panic.rs:146:14
  27:   0x189c1c - std::rt::lang_start_internal::{closure#2}
                       at /checkout/library/std/src/rt.rs:148:48
  28:   0xa2f696 - std::panicking::r#try::do_call::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
                       at /checkout/library/std/src/panicking.rs:552:40
  29:   0xa2f02f - std::panicking::r#try::<isize, {closure@std::rt::lang_start_internal::{closure#2}}>
                       at /checkout/library/std/src/panicking.rs:516:19
  30:    0x7f680 - std::panic::catch_unwind::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
  31:   0x189c1d - std::rt::lang_start_internal
                       at /checkout/library/std/src/rt.rs:148:20
  32:   0x189e0d - std::rt::lang_start::<()>
                       at /checkout/library/std/src/rt.rs:165:17
---
   |
   = note: inside `std::sys::pal::windows::abort_internal` at /checkout/library/std/src/sys/pal/windows/mod.rs:340:5: 340:30
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:798:9: 798:37
   = note: inside closure at /checkout/library/std/src/panicking.rs:649:13: 655:14
   = note: inside `std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys_common/backtrace.rs:171:18: 171:21
   = note: inside `core::panicking::panic_nounwind_nobacktrace` at /checkout/library/core/src/panicking.rs:163:5: 163:90
   = note: inside `core::panicking::panic_in_cleanup` at /checkout/library/core/src/panicking.rs:256:5: 256:71
note: inside `main`
  --> tests/fail/panic/double_panic.rs:13:1
---



FAILED TEST: tests/fail/function_calls/exported_symbol_bad_unwind2.rs (revision `definition`)
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-msvc" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui/tests/fail/function_calls" "tests/fail/function_calls/exported_symbol_bad_unwind2.rs" "--cfg=definition" "--edition" "2021"
error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/function_calls/exported_symbol_bad_unwind2.definition.stderr` to the actual output
--- tests/fail/function_calls/exported_symbol_bad_unwind2.definition.stderr
+++ <stderr output>
---
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /checkout/library/core/src/panicking.rs:155:5:
panic in a function that cannot unwind
stack backtrace:
   0:   0xbee4b3 - std::backtrace_rs::backtrace::miri::trace_unsynchronized::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:99:5
   1:   0xbee055 - std::backtrace_rs::backtrace::miri::trace::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:62:14
   2:   0xbec9d2 - std::backtrace_rs::backtrace::trace_unsynchronized::<{closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
   3:   0xbc35f6 - std::sys_common::backtrace::_print_fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:68:5
   4:   0xbc232b - <std::sys_common::backtrace::_print::DisplayBacktrace as std::fmt::Display>::fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:44:22
---
  12:   0x19a0cd - std::panicking::rust_panic_with_hook
                       at /checkout/library/std/src/panicking.rs:779:13
  13:   0x198d40 - std::panicking::begin_panic_handler::{closure#0}
                       at /checkout/library/std/src/panicking.rs:649:13
  14:   0xbc4460 - std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>
  15:   0x198c3d - std::panicking::begin_panic_handler
                       at /checkout/library/std/src/panicking.rs:645:5
  16:   0x15db3b - core::panicking::panic_nounwind_fmt::runtime
                       at /checkout/library/core/src/panicking.rs:110:18
---
  28:    0x806de - std::panic::catch_unwind::<&dyn std::ops::Fn() -> i32 + std::marker::Sync + std::panic::RefUnwindSafe, i32>
                       at /checkout/library/std/src/panic.rs:146:14
  29:   0x1761bf - std::rt::lang_start_internal::{closure#2}
                       at /checkout/library/std/src/rt.rs:148:48
  30:   0x198035 - std::panicking::r#try::do_call::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
                       at /checkout/library/std/src/panicking.rs:552:40
  31:   0x1979c6 - std::panicking::r#try::<isize, {closure@std::rt::lang_start_internal::{closure#2}}>
                       at /checkout/library/std/src/panicking.rs:516:19
  32:    0x806fc - std::panic::catch_unwind::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
  33:   0x1761c4 - std::rt::lang_start_internal
                       at /checkout/library/std/src/rt.rs:148:20
  34:   0x1763b7 - std::rt::lang_start::<()>
                       at /checkout/library/std/src/rt.rs:165:17
---
   |
   = note: inside `std::sys::pal::windows::abort_internal` at /checkout/library/std/src/sys/pal/windows/mod.rs:340:5: 340:30
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:798:9: 798:37
   = note: inside closure at /checkout/library/std/src/panicking.rs:649:13: 655:14
   = note: inside `std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys_common/backtrace.rs:171:18: 171:21
   = note: inside `core::panicking::panic_nounwind` at /checkout/library/core/src/panicking.rs:155:5: 155:91
   = note: inside `core::panicking::panic_cannot_unwind` at /checkout/library/core/src/panicking.rs:241:5: 241:61
note: inside `nounwind`
  --> tests/fail/function_calls/exported_symbol_bad_unwind2.rs:11:1
  --> tests/fail/function_calls/exported_symbol_bad_unwind2.rs:11:1
   |
LL | / extern "C-unwind" fn nounwind() {
LL | | }
   | |_^
note: inside `main`
  --> tests/fail/function_calls/exported_symbol_bad_unwind2.rs:20:14
---



FAILED TEST: tests/fail/function_calls/exported_symbol_bad_unwind2.rs (revision `both`)
command: MIRI_ENV_VAR_TEST="0" MIRI_TEMP="/tmp" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/miri" "--error-format=json" "-Dwarnings" "-Dunused" "-Ainternal_features" "-Zui-testing" "--target" "i686-pc-windows-msvc" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-tools/ui/tests/fail/function_calls" "tests/fail/function_calls/exported_symbol_bad_unwind2.rs" "--cfg=both" "--edition" "2021"
error: actual output differed from expected
error: actual output differed from expected
Execute `./miri test --bless` to update `tests/fail/function_calls/exported_symbol_bad_unwind2.both.stderr` to the actual output
--- tests/fail/function_calls/exported_symbol_bad_unwind2.both.stderr
 thread 'main' panicked at $DIR/exported_symbol_bad_unwind2.rs:LL:CC:
 explicit panic
... 6 lines skipped ...
   --> RUSTLIB/std/src/sys/pal/PLATFORM/mod.rs:LL:CC
---
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /checkout/library/core/src/panicking.rs:155:5:
panic in a function that cannot unwind
stack backtrace:
   0:   0xbee4b3 - std::backtrace_rs::backtrace::miri::trace_unsynchronized::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:99:5
   1:   0xbee055 - std::backtrace_rs::backtrace::miri::trace::<&mut {closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
                       at /checkout/library/std/src/../../backtrace/src/backtrace/miri.rs:62:14
   2:   0xbec9d2 - std::backtrace_rs::backtrace::trace_unsynchronized::<{closure@std::sys_common::backtrace::_print_fmt::{closure#1}}>
   3:   0xbc35f6 - std::sys_common::backtrace::_print_fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:68:5
   4:   0xbc232b - <std::sys_common::backtrace::_print::DisplayBacktrace as std::fmt::Display>::fmt
                       at /checkout/library/std/src/sys_common/backtrace.rs:44:22
---
  12:   0x19a0cd - std::panicking::rust_panic_with_hook
                       at /checkout/library/std/src/panicking.rs:779:13
  13:   0x198d40 - std::panicking::begin_panic_handler::{closure#0}
                       at /checkout/library/std/src/panicking.rs:649:13
  14:   0xbc4460 - std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>
  15:   0x198c3d - std::panicking::begin_panic_handler
                       at /checkout/library/std/src/panicking.rs:645:5
  16:   0x15db3b - core::panicking::panic_nounwind_fmt::runtime
                       at /checkout/library/core/src/panicking.rs:110:18
---
  28:    0x806de - std::panic::catch_unwind::<&dyn std::ops::Fn() -> i32 + std::marker::Sync + std::panic::RefUnwindSafe, i32>
                       at /checkout/library/std/src/panic.rs:146:14
  29:   0x1761bf - std::rt::lang_start_internal::{closure#2}
                       at /checkout/library/std/src/rt.rs:148:48
  30:   0x198035 - std::panicking::r#try::do_call::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
                       at /checkout/library/std/src/panicking.rs:552:40
  31:   0x1979c6 - std::panicking::r#try::<isize, {closure@std::rt::lang_start_internal::{closure#2}}>
                       at /checkout/library/std/src/panicking.rs:516:19
  32:    0x806fc - std::panic::catch_unwind::<{closure@std::rt::lang_start_internal::{closure#2}}, isize>
  33:   0x1761c4 - std::rt::lang_start_internal
                       at /checkout/library/std/src/rt.rs:148:20
  34:   0x1763b7 - std::rt::lang_start::<()>
                       at /checkout/library/std/src/rt.rs:165:17
---
   |
   = note: inside `std::sys::pal::windows::abort_internal` at /checkout/library/std/src/sys/pal/windows/mod.rs:340:5: 340:30
   = note: inside `std::panicking::rust_panic_with_hook` at /checkout/library/std/src/panicking.rs:798:9: 798:37
   = note: inside closure at /checkout/library/std/src/panicking.rs:649:13: 655:14
   = note: inside `std::sys_common::backtrace::__rust_end_short_backtrace::<{closure@std::panicking::begin_panic_handler::{closure#0}}, !>` at /checkout/library/std/src/sys_common/backtrace.rs:171:18: 171:21
   = note: inside `core::panicking::panic_nounwind` at /checkout/library/core/src/panicking.rs:155:5: 155:91
   = note: inside `core::panicking::panic_cannot_unwind` at /checkout/library/core/src/panicking.rs:241:5: 241:61
note: inside `nounwind`
  --> tests/fail/function_calls/exported_symbol_bad_unwind2.rs:11:1
  --> tests/fail/function_calls/exported_symbol_bad_unwind2.rs:11:1
   |
LL | / extern "C-unwind" fn nounwind() {
LL | | }
   | |_^
note: inside `main`
  --> tests/fail/function_calls/exported_symbol_bad_unwind2.rs:20:14
```

</details>

--- a/library/panic_abort/src/lib.rs
+++ b/library/panic_abort/src/lib.rs
@@ -75,16 +75,15 @@ pub unsafe fn __rust_start_panic(_payload: &mut dyn PanicPayload) -> u32 {
                 const FAST_FAIL_FATAL_APP_EXIT: usize = 7;
                 cfg_if::cfg_if! {
                     if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
-                        core::arch::asm!("int $$0x29", in("ecx") FAST_FAIL_FATAL_APP_EXIT);
+                        core::arch::asm!("int $$0x29", in("ecx") FAST_FAIL_FATAL_APP_EXIT, options(noreturn, nostack));
                     } else if #[cfg(all(target_arch = "arm", target_feature = "thumb-mode"))] {
-                        core::arch::asm!(".inst 0xDEFB", in("r0") FAST_FAIL_FATAL_APP_EXIT);
+                        core::arch::asm!(".inst 0xDEFB", in("r0") FAST_FAIL_FATAL_APP_EXIT, options(noreturn, nostack));
                     } else if #[cfg(target_arch = "aarch64")] {
-                        core::arch::asm!("brk 0xF003", in("x0") FAST_FAIL_FATAL_APP_EXIT);
+                        core::arch::asm!("brk 0xF003", in("x0") FAST_FAIL_FATAL_APP_EXIT, options(noreturn, nostack));
                     } else {
                         core::intrinsics::abort();
                     }
                 }
-                core::intrinsics::unreachable();
             }
         } else if #[cfg(target_os = "teeos")] {
             mod teeos {

--- a/library/std/src/sys/pal/windows/c/bindings.txt
+++ b/library/std/src/sys/pal/windows/c/bindings.txt
@@ -2481,6 +2481,7 @@ Windows.Win32.System.SystemInformation.SYSTEM_INFO
 Windows.Win32.System.SystemServices.DLL_PROCESS_DETACH
 Windows.Win32.System.SystemServices.DLL_THREAD_DETACH
 Windows.Win32.System.SystemServices.EXCEPTION_MAXIMUM_PARAMETERS
+Windows.Win32.System.SystemServices.FAST_FAIL_FATAL_APP_EXIT
 Windows.Win32.System.SystemServices.IO_REPARSE_TAG_MOUNT_POINT
 Windows.Win32.System.SystemServices.IO_REPARSE_TAG_SYMLINK
 Windows.Win32.System.Threading.ABOVE_NORMAL_PRIORITY_CLASS

--- a/library/std/src/sys/pal/windows/c/windows_sys.rs
+++ b/library/std/src/sys/pal/windows/c/windows_sys.rs
@@ -3086,6 +3086,7 @@ pub type FACILITY_CODE = u32;
 pub const FACILITY_NT_BIT: FACILITY_CODE = 268435456u32;
 pub const FALSE: BOOL = 0i32;
 pub type FARPROC = ::core::option::Option<unsafe extern "system" fn() -> isize>;
+pub const FAST_FAIL_FATAL_APP_EXIT: u32 = 7u32;
 #[repr(C)]
 pub struct FD_SET {
     pub fd_count: u32,

--- a/library/std/src/sys/pal/windows/mod.rs
+++ b/library/std/src/sys/pal/windows/mod.rs
@@ -321,25 +321,26 @@ pub fn dur2timeout(dur: Duration) -> c::DWORD {
 ///
 /// This is the same implementation as in libpanic_abort's `__rust_start_panic`. See
 /// that function for more information on `__fastfail`
-#[allow(unreachable_code)]
+#[cfg(not(miri))] // inline assembly does not work in Miri
 pub fn abort_internal() -> ! {
-    #[allow(unused)]
-    const FAST_FAIL_FATAL_APP_EXIT: usize = 7;
-    #[cfg(not(miri))] // inline assembly does not work in Miri
     unsafe {
         cfg_if::cfg_if! {
             if #[cfg(any(target_arch = "x86", target_arch = "x86_64"))] {
-                core::arch::asm!("int $$0x29", in("ecx") FAST_FAIL_FATAL_APP_EXIT);
-                crate::intrinsics::unreachable();
+                core::arch::asm!("int $$0x29", in("ecx") c::FAST_FAIL_FATAL_APP_EXIT, options(noreturn, nostack));
             } else if #[cfg(all(target_arch = "arm", target_feature = "thumb-mode"))] {
-                core::arch::asm!(".inst 0xDEFB", in("r0") FAST_FAIL_FATAL_APP_EXIT);
-                crate::intrinsics::unreachable();
+                core::arch::asm!(".inst 0xDEFB", in("r0") c::FAST_FAIL_FATAL_APP_EXIT, options(noreturn, nostack));
             } else if #[cfg(target_arch = "aarch64")] {
-                core::arch::asm!("brk 0xF003", in("x0") FAST_FAIL_FATAL_APP_EXIT);
-                crate::intrinsics::unreachable();
+                core::arch::asm!("brk 0xF003", in("x0") c::FAST_FAIL_FATAL_APP_EXIT, options(noreturn, nostack));
+            } else {
+                core::intrinsics::abort();
             }
         }
     }
+}
+
+// miri is sensitive to changes here so check that miri is happy if touching this
+#[cfg(miri)]
+pub fn abort_internal() -> ! {
     crate::intrinsics::abort();
 }
 

